### PR TITLE
chore(eap-spans): Create a new materialized view for attributes based on the spans v2 table

### PR DESCRIPTION
**Pull request by phacops**
- Add retention_days to the partition expression so we can drop parts and not do TTLMerge and still remove rows on time
- Convert `duration_micro` to milliseconds before inserting in the attributes table
- Feed the materialize views from `eap_spans_2_local` instead of `eap_spans_local`

### CODE REVIEW

**Review by volokluev — COMMENTED** [review]
Is there a reason you need to drop and create in the same migration?

**Comment by volokluev** [comment]
Also is there anything materially different about these tables I should pay attention to?

**Comment by phacops** [comment]
> Is there a reason you need to drop and create in the same migration?

No reason, I'll remove it.

> Also is there anything materially different about these tables I should pay attention to?

- Add retention_days to the partition expression so we can drop parts and not do TTLMerge and still remove rows on time
- Convert `duration_micro` to milliseconds before inserting in the attributes table
- Feed the materialize views from `eap_spans_2_local` instead of `eap_spans_local`

File: snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                view_name=self.new_num_mv,
+                columns=self.num_columns,
+                destination_table_name=self.new_num_local_table,
+                target=OperationTarget.LOCAL,
+                query=f"""
+SELECT
+    organization_id,
+    project_id,
+    attrs.1 as attr_key,
+    attrs.2 as attr_min_value,
```
[the remark this answers is not in the document]
Comment (phacops) [reply snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py:166]: Using `SimpleAggregationFunction` columns make it unnecessary to have `minState`. It needs a value, not a function.

**Review by colin-sentry — APPROVED** [review]

File: snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    new_str_local_table = "spans_str_attrs_3_local"
+    new_str_dist_table = "spans_str_attrs_3_dist"
+    str_columns: Sequence[Column[Modifiers]] = [
+        Column("organization_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("attr_key", String(modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column("attr_value", String(modifiers=Modifiers(codecs=["ZSTD(1)"]))),
+        Column(
+            "timestamp",
+            DateTime(modifiers=Modifiers(codecs=["DoubleDelta", "ZSTD(1)"])),
```
Comment (colin-sentry) [inline snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py:43]: was doubledelta recommended against? i forget what altinity consultant said

Comment (phacops) [reply snuba/snuba_migrations/events_analytics_platform/0017_span_attribute_table_v3.py:43]: Not for timestamps, that's the only place where he recommended it.

**Review by evanh — APPROVED** [review]

**Pull request merged**